### DISCUSSION TITLE
Scan all videos from multiple creators and automate quota usage

### DISCUSSION THREAD

**Discussion by MoralCode** [ThioJoe/YT-Spammer-Purge#465] [Ideas]
Hello, Thanks for making this script!

After playing with this a bit, i was thinking it would be nice to be able to set up this script to automatically process a large set of videos (i.e. all videos from a large list of creators) across however many days/scheduled script runs it takes to get enough API calls allocated to process them all. This seems like it would break down into multiple features:

- detecting when the API quota is hit (seems like this cold also indirectly solve #461 since detection is half of making a better error message)
- being able to persist un-processed jobs/a list of api calls to be made to disk or something so it can save anything there wasnt enough quota for until the next day's scheduled run (maybe with a config flag to automatically resume any detected in-progress batch jobs without a prompt?)
- being able to process a list of creators sequentially in an automated way (maybe this is already possible?)
- adding an "all" setting when selecting how many recent videos to process. this might be a neat way to kick off a "batch/bulk job" (see point #2) automatically

What do people think of this idea?

**Comment by ThioJoe** [comment]
Well for the one point, you can actually use a log for deletion, and only delete a partial number of the comments, and it will save your progress and you can continue later. The other problem you'll run up against real quick is an API limit. Fetching 500 most recent videos for example would use up your entire quota for a project, for example (before scanning a single comment).

At some point I'll probably add a way to "queue up" multiple google cloud projects to work around the api limit though.

**Reply by MoralCode** [reply]
do you think it would be possible to do things in stages or cache anything you get from the API so that it can be picked up in later runs? i.e. say i used my whole quota for those 500 fetches, id want to be able to run the script the next day and have it either continue fetching more or start grabbing comments from those videos that it already fetched and retrieved from a cache. 

Do you imagine this kind of thing as being part of this project in the short to medium term?

**Comment by djsv23** [comment]
With the docker container set up now, you can create a config file for each creator and mount each into a separate instance of the container. Using docker-compose, you can define all of this ahead of time and then just with one "docker-compose up" command it will run the script for all of the creators you've defined configs for. 

This doesn't really handle the automatic API monitoring piece of your idea, but it gets quite close and with a little experimentation you can find out how many creators/videos you can get through in a day without reaching the limit. This approach also makes it pretty simple to set up multiple GCP projects and pass different client_secret.json's into each container to effectively multiply your limits.
